### PR TITLE
TSK-1783: Updated CompleteTaskAccTest to use Test-API

### PR DESCRIPTION
**Pull request by SebastianRoseneck**
---
 
### For the submitter:
- [ ] I updated the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) and will supply links to the specific files
- [x] I did not update the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview)
- [ ] I included a link to the [SonarCloud branch analysis](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1019969636/SonarCloud+Integration)
- [ ] I added a description of changes on the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I did not update the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I put the ticket in review
- [ ] After integration of the pull request, I verified our [bluemix test environment](http://taskana.mybluemix.net/taskana) is not broken

### Verified by the reviewer:
- [ ] Commit message format → TSK-XXX: Your commit message.
- [ ] Submitter's update to [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) is sufficient
- [ ] SonarCloud analysis meets our standards
- [ ] Update of the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana) reflects changes
- [ ] PR fulfills the ticket
- [ ] Edge cases and unwanted side effects are tested
- [ ] Readability

### CODE REVIEW

File: lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java:715]: why are you creating the task as an admin?

File: lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java:78]: in my opinion not necessary anymore, since we've created a task which is claimed.

File: lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java:76]: rename to e.g. `claimedTask` to clarify that the task is already claimed

File: lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java:91]: rename to e.g. claimedTask to clarify that the task is already claimed

File: lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java:106]: rename to e.g. claimedTask to clarify that the task is already claimed

File: lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java:119]: rename to e.g. claimedTask to clarify that the task is already claimed

File: lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java:130]: Idea: change the return type of `createTaskClaimedByUser` to `TaskBuilder`.
Then you can just write Task claimedBydifferentUser ?= createTaskClaimedByUser().owner("other").buildAndStore([...])

This means that all other calls have to be `Task claimedTask = createTaskClaimedByUser().build()` . What do you think?

File: lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java:713]: We can reuse `createDefaultTask`

**Review by mustaphazorgati — CHANGES_REQUESTED** [review]
This is just a partial review. I needed a break from my other stuff..

This needs approval from someone else ^^

**Comment by mustaphazorgati** [comment]
@SebastianRoseneck please rebase this PR. thanks :)

File: lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java:83]: I know you didn't modify this, but maybe you're willing to make an adjustment here ^^

Instead of asserting modified is not created I'd personally assert that completed = modify != null != created.

File: lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java:95]: Same thing as above: 

> I know you didn't modify this, but maybe you're willing to make an adjustment here ^^
Instead of asserting modified is not created I'd personally assert that completed = modify != null != created.

File: lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java:110]: Same thing as above: 

> I know you didn't modify this, but maybe you're willing to make an adjustment here ^^
Instead of asserting modified is not created I'd personally assert that completed = modify != null != created.

File: lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java:157]: General note regarding the exceptions:
Since TSK-1647 our exceptions contain some information.
Should we modify the assert statements accordingly in this ticket?

File: lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java:207]: Since we create the task this is not necessary anymore :)

File: lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java:213]: Please inject the CurrentUserContext, when it's used here :)

File: lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java:227]: not necessary, since we've created the task.

File: lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java:229]: truncate is not necessary :)

File: lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java:258]: Please extract the lambda statement, so that we use the given, when, then theorem :)
And don't forget to add empty lines between each block. Thanks!

File: lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java:271]: > Please extract the lambda statement, so that we use the given, when, then theorem :)
And don't forget to add empty lines between each block. Thanks!

File: lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java:283]: > Please extract the lambda statement, so that we use the given, when, then theorem :)
And don't forget to add empty lines between each block. Thanks!

File: lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java:294]: not necessary

File: lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java:314]: Not your fault: Please change the this
``` 
assertThat(taskAfter.getModified().isAfter(before)).isTrue();
```
to this
```
assertThat(taskAfter.getModified()).isAfter(before);
```

**Review by mustaphazorgati — CHANGES_REQUESTED** [review]
Hey @SebastianRoseneck,

this is a pretty long testfile.
When you're done with the changes I'd prefer looking at this together so that we don't miss anything.

File: lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java
Comment (SebastianRoseneck) [reply lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java]: Do you mean the assert in line 78 or the entire test?

Comment (mustaphazorgati) [reply lib/taskana-core/src/test/java/acceptance/task/CompleteTaskAccTest.java]: The assert

File: lib/taskana-core/src/test/java/acceptance/task/complete/CompleteTaskAccTest.java
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/complete/CompleteTaskAccTest.java:82]: since the assert statements two lines below already contains `isNotNull` this is not necessary.

File: lib/taskana-core/src/test/java/acceptance/task/complete/CompleteTaskAccTest.java
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/complete/CompleteTaskAccTest.java:115]: since the assert statements three lines below already contains `isNotNull` this is not necessary.

File: lib/taskana-core/src/test/java/acceptance/task/complete/CompleteTaskAccTest.java
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/complete/CompleteTaskAccTest.java:116]: since the assert statements two lines below already contains `isNotNull` this is not necessary.

File: lib/taskana-core/src/test/java/acceptance/task/complete/CompleteTaskAccTest.java
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/complete/CompleteTaskAccTest.java:115]: since the assert statements three lines below already contains `isNotNull` this is not necessary.

File: lib/taskana-core/src/test/java/acceptance/task/complete/CompleteTaskAccTest.java
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/complete/CompleteTaskAccTest.java:98]: since the assert statements two lines below already contains `isNotNull` this is not necessary.

File: lib/taskana-core/src/test/java/acceptance/task/complete/CompleteTaskAccTest.java
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/complete/CompleteTaskAccTest.java:143]: since the assert statements two lines below already contains `isNotNull` this is not necessary.

File: lib/taskana-core/src/test/java/acceptance/task/complete/CompleteTaskAccTest.java
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/complete/CompleteTaskAccTest.java:159]: why did you adjust the `completed` assert (as I requested) only sometimes? E.g. here it's missing ^^

File: lib/taskana-core/src/test/java/acceptance/task/complete/CompleteTaskAccTest.java
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/complete/CompleteTaskAccTest.java:197]: This is long and hard to read :/ 
Let's find a simpler way to do the same thing :)

This might help: https://joel-costigliola.github.io/assertj/core-8/api/org/assertj/core/api/Assertions.html#catchThrowableOfType-org.assertj.core.api.ThrowableAssert.ThrowingCallable-java.lang.Class-

Alternatives could be some suggestions from this question: https://stackoverflow.com/questions/64541192/assertthatthrownby-check-field-on-custom-exception

File: lib/taskana-core/src/test/java/acceptance/task/complete/CompleteTaskAccTest.java
Comment (mustaphazorgati) [inline lib/taskana-core/src/test/java/acceptance/task/complete/CompleteTaskAccTest.java:751]: when we want to use our API correctly we have to set the claimed timestamp aswell :)

**Review by mustaphazorgati — CHANGES_REQUESTED** [review]
This is just a partial. I g2g for now.

**Review by mustaphazorgati — APPROVED** [review]

**Review by mustaphazorgati — APPROVED** [review]
Please rebase and move this file to the new `taskana-core-test` module, after #1928 has been merged.

**Review by mustaphazorgati — DISMISSED** [review]

**Review by mustaphazorgati — APPROVED** [review]

**Pull request merged**
